### PR TITLE
Report accounts

### PR DESCRIPTION
**Pull request by lomamech**
Report Accounts
- Implementation of client side 
- Implementation of API and Report PDF Design
- Implementation of integration test ans E2E test

@sfount  @mbayopanda

### CODE REVIEW

File: client/src/i18n/fr.json [hunk 1]
```diff
@@ -684,7 +684,7 @@
     },
     "SELECT": {
       "EMPTY"              : "Vide",
-      "ACCOUNT"            : "Sélectionner compte",
+      "ACCOUNT"            : "Sélectionner un compte",
```
Comment (jniles) [inline client/src/i18n/fr.json:689]: :+1:

File: client/src/partials/reports/modals/report_accounts.config.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+function ReportAccountsConfigController($state, ModalInstance, Accounts, Notify, Languages, SavedReports, reportDetails) {
+  var vm = this;
+  var report = reportDetails;
+
+  // expose to the view
+  vm.generate = generate;
+  vm.cancel = ModalInstance.dismiss;
+  vm.report = report;
+
+  Accounts.read(null, { detailed: 1, is_auxiliary: 0})
```
Comment (jniles) [inline client/src/partials/reports/modals/report_accounts.config.js:34]: You don't need `is_auxiliary` here...

File: client/src/partials/reports/modals/report_accounts.config.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      accounts.forEach(function (account) {
+        account.hrlabel = account.number + ' ' + account.label;
+      });
+
+      vm.accounts = accounts;
+    })
+    .catch(Notify.errorHandler);
+
+  function generate(form) {
+    var url = 'reports/finance/report_accounts';
```
Comment (jniles) [inline client/src/partials/reports/modals/report_accounts.config.js]: This report is for a single account, correct?  I propose that this URL should be `reports/finance/account`.

File: client/src/partials/reports/modals/report_accounts.config.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    if (form.$invalid) { return; }
+
+    var options = {
+      account_id: vm.account.id,
+      account_label: vm.account.label,
```
Comment (jniles) [inline client/src/partials/reports/modals/report_accounts.config.js]: Do you actually need the parameters `account_label` and `account_number`?

File: server/controllers/finance/reports/reportAccounts/report.handlebars [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{{> head title="PATIENT_RECORDS.FINANCIAL_ACTIVITY.REPORT_TITLE" }}
```
Comment (jniles) [inline server/controllers/finance/reports/reportAccounts/report.handlebars]: I think this title is wrong.

File: server/controllers/finance/reports/reportAccounts/report.handlebars [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  <h3 class="text-center text-uppercase"><strong>{{translate "REPORT.ACCOUNT"}}</strong></h3>
+  <h4 class="text-center"><strong>{{ title.accountNumber }}</strong></h4>
+  <h4 class="text-center text-uppercase"><strong>{{ title.accountLabel}}</strong></h4>
+  
+  <section>
+    <table class="table table-condensed table-striped table-report table-bordered">
+      <thead>
+        <tr class="text-capitalize text-center" style="background-color: #ddd;">
+          <th style="width: 10%;">{{translate "TABLE.COLUMNS.INVOICE_ID" }}</th>
```
Comment (jniles) [inline server/controllers/finance/reports/reportAccounts/report.handlebars]: Only class 4 accounts will have invoice IDs.  I propose you remove this column for a generic account report.

**Review by jniles — CHANGES_REQUESTED** [review]
@lomamech, the code is pretty good.  I've made a few comments to improve.

For functionality, I propose you:
1. Remove the `Invoice ID`, since most accounts will never have an invoice against them.  You should remove it from the SQL as well.
2. Add an option to include the posting journal that is `false` by default.  Currently, both the General Ledger and Posting Journal is shown by default.  It should be just the General Ledger by default, with an option to enable the Posting Journal.

File: client/src/partials/reports/modals/report_accounts.config.js (hunk 4 above)
Comment (lomamech) [reply client/src/partials/reports/modals/report_accounts.config.js]: Yes, For display in the PDF report

Comment (jniles) [reply client/src/partials/reports/modals/report_accounts.config.js]: Ah, I see it now.  :+1:

**Comment by lomamech** [comment]
@jniles
This branch is already for be merged

**Comment by jniles** [comment]
@lomamech, I'll take a look.  Thanks!

**Comment by jniles** [comment]
LGTM :+1:

**Pull request merged**